packages/sqlite/src/index.ts

```diff
@@ -118,115 +92,91 @@ class DatabaseConnectionImplementation implements DatabaseConnection {
 
   async tx<T>(fn: (db: DatabaseTransaction) => Promise<T>): Promise<T> {
     return this._mutex.writeLock(async () => {
-      await new Promise<void>((resolve, reject) => {
-        this._database.run('BEGIN', (err) => {
-          if (err) reject(err);
-          else resolve();
-        });
-      });
+      // TODO extract those as class variables
+      // however I do not know how to get
+      this._begin.run();
       try {
-        const result = fn(
+        const result = await fn(
           new DatabaseTransactionImplementation(this._database),
         );
-        await new Promise<void>((resolve, reject) => {
-          this._database.run('COMMIT', (err) => {
-            if (err) reject(err);
-            else resolve();
-          });
-        });
+        this._commit.run();
```

Comment:
Does `.run()` return a Promise, or is it synchronous? If it returns a promise we'll need to `await` here.

Reply:
No, `better-sqlite3` is all synchronous

Reply:
Ah, I hadn't understood that when we initially discussed this. I think based on that we should probably make this a new package called `@databases/sqlite-sync` as it will have very different performance characteristics, and a purely synchronous client can have a simpler API where all methods return values directly, rather than promises.

Assuming better-sqlite3 is a clear improvement over the existing db driver, we can then replace `@databases/sqlite` with a totally re-written (but API compatible) library that makes use of a worker thread (https://github.com/WiseLibs/better-sqlite3/blob/master/docs/threads.md) to provide a truly asynchronous API on top of better-sqlite3.